pageserver/src/tenant/layer_map.rs

```diff
@@ -20,18 +20,18 @@ use num_traits::{Bounded, Num, Signed};
 use rstar::{RTree, RTreeObject, AABB};
 use std::cmp::Ordering;
 use std::collections::VecDeque;
-use std::ops::Range;
 use std::ops::{Add, Div, Mul, Neg, Rem, Sub};
+use std::ops::{Deref, Range};
 use std::sync::Arc;
 use tracing::*;
 use utils::lsn::Lsn;
 
-use super::storage_layer::{InMemoryLayer, Layer};
+use super::storage_layer::{InMemoryLayer, Layer, PersistentLayer};
 
 ///
 /// LayerMap tracks what layers exist on a timeline.
 ///
-pub struct LayerMap<L: ?Sized> {
+pub struct LayerMap<L = PersistentLayer> {
```

Comment:
I don't understand what `L = PersistentLayer` means. Does that make `L` just an alias for `PersistentLayer`? But the benchmark code uses `LayerMap<DummyMapLayer>`. How is `LayerMap<DummyMapLayer>` possible, with the constraint that `L = PeristentLayer`?

Reply:
It's not a constraint, it's a default generic parameters: https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#default-generic-type-parameters-and-operator-overloading

`LayerMap<L = PersistentLayer>` means "`LayerMap` is generic, use `PersistentLayer` as a generic value if nothing else is specified explicitly"

Reply:
gotcha, thanks!